lib/util/http/gitlab.spec.ts

```diff
@@ -68,6 +71,21 @@ describe('util/http/gitlab', () => {
     expect(trace).toHaveLength(3);
     expect(trace).toMatchSnapshot();
   });
+
+  it('supports different datasources', async () => {
+    const gitlabApiDatasource = new GitlabHttp(GitlabReleasesDatasource.id);
+    hostRules.add({ hostType: PLATFORM_TYPE_GITLAB, token: 'abc' });
+    hostRules.add({
+      hostType: GitlabReleasesDatasource.id,
+      token: 'def',
+    });
+    httpMock.scope(gitlabApiHost).get('/api/v4/some-url').reply(200);
```

Comment:
```suggestion
    httpMock.scope(gitlabApiHost, { reqheaders: {authorization: 'Bearer def'}}).get('/api/v4/some-url').reply(200);
```

use nock header matching

https://github.com/nock/nock#specifying-headers